### PR TITLE
Wassaf/404 add search icon

### LINKED ISSUE #404: All page headers should have search icon
... except for the home page and the search page, where the search icon would be redundant in the header. 

Course pages already have the search icon in the header, for example, https://ocwnext-rc.odl.mit.edu/courses/16-885j-aircraft-systems-engineering-fall-2004/

#### Acceptance Criteria:

- [ ] about page
- [ ] educator page
- [ ] contact page
- [ ] newsletter page
- [ ] testimonial pages
- [ ] privacy and terms of service page, e.g. https://ocw-draft-qa.global.ssl.fastly.net/pages/privacy-and-terms-of-use/

#### Out of Scope

Course pages already have the search icon in the header, for example, https://ocwnext-rc.odl.mit.edu/courses/16-885j-aircraft-systems-engineering-fall-2004/

### PR DESCRIPTION
**Pull request by Wassaf-Shahzad**
#### Pre-Flight checklist

- [x] Screenshots and design review for any changes that affect layout or styling
  - [x] Desktop screenshots
- [x] Testing
   - [x] Changes have been manually tested

#### What are the relevant tickets?
Closes #404

#### What's this PR do?
- adds search icon to required pages

#### How should this be manually tested?
- Verify all pages have search icon
- Verify home page and search page dont have search icon

#### Screenshots (if appropriate)
Screenshot 2022-02-03 at 1 28 33 PM

### CODE REVIEW

**Comment by pdpinch** [comment]
I'd like to get @gumaerc's input on this approach before it's merged.

**Review by mbertrand — APPROVED** [review]
Looks great 👍

**Review by gumaerc — CHANGES_REQUESTED** [review]
I think we should go with a different approach here.  First of all, partials like this that will likely return the same thing every time should be called with `partialCached`. 
 Secondly, removing the `.` as an argument means the `header.html` partial does not receive a context object.  This means that the partial template will not have access to the calling page or any of the site wide data.  This also creates a situation where the header partial will be receiving inconsistent arguments.  If you search `ocw-hugo-themes` for `{{ partialCached "header" . }}` you'll see that there are a lot of places where it's called this way.  A better approach here would be to store this variable in the front matter of pages where it needs to be set.  For instance, in `www/content/search/_index.md` you could set:

``` 
---
title: Search
type: search
notRenderSearchIcon: true
---
```

Then, at `www/content` you could create an `_index.md` file with:

```
---
notRenderSearchIcon: true
---
```

The code `{{if ne (isset . "notRenderSearchIcon") true}}` also does not actually check the value of this variable.  It only checks for the existence of it.  Regardless, the search icon may be the only thing in `extraheader` right now, but we might want to add more to this partial in the future and then this flag would hide that as well.  Instead, if we preserve passing only the context variable, in `extraheader.html` we can do something like this in combination with the changes I described above:

```
{{ define "extraheader" }}
{{ $notRenderSearchIcon := index .Params "notRenderSearchIcon" | default false }}
{{ if not $notRenderSearchIcon }}
<a class="nav-link search-icon pr-6" href="/search">
  <i class="material-icons">search</i>
</a>
{{ end }}
{{ end }}
```

This way, any particular type of page can be marked to not render the search icon in the content itself and you can default it to false easily.

**Comment by Wassaf-Shahzad** [comment]
> I think we should go with a different approach here. First of all, partials like this that will likely return the same thing every time should be called with `partialCached`. Secondly, removing the `.` as an argument means the `header.html` partial does not receive a context object. This means that the partial template will not have access to the calling page or any of the site wide data. This also creates a situation where the header partial will be receiving inconsistent arguments. If you search `ocw-hugo-themes` for `{{ partialCached "header" . }}` you'll see that there are a lot of places where it's called this way. A better approach here would be to store this variable in the front matter of pages where it needs to be set. For instance, in `www/content/search/_index.md` you could set:
> 
> ```
> ---
> title: Search
> type: search
> notRenderSearchIcon: true
> ---
> ```
> 
> Then, at `www/content` you could create an `_index.md` file with:
> 
> ```
> ---
> notRenderSearchIcon: true
> ---
> ```
> 
> The code `{{if ne (isset . "notRenderSearchIcon") true}}` also does not actually check the value of this variable. It only checks for the existence of it. Regardless, the search icon may be the only thing in `extraheader` right now, but we might want to add more to this partial in the future and then this flag would hide that as well. Instead, if we preserve passing only the context variable, in `extraheader.html` we can do something like this in combination with the changes I described above:
> 
> ```
> {{ define "extraheader" }}
> {{ $notRenderSearchIcon := index .Params "notRenderSearchIcon" | default false }}
> {{ if not $notRenderSearchIcon }}
> <a class="nav-link search-icon pr-6" href="/search">
>   <i class="material-icons">search</i>
> </a>
> {{ end }}
> {{ end }}
> ```
> 
> This way, any particular type of page can be marked to not render the search icon in the content itself and you can default it to false easily.

I went over this approach and discovered a few things. Caching a partial usually saves its first instance and renders that one every where else. Which causes problems in rendering the `extraheader` dynamically since if you visit the home page first and the Header is rendered without  `extraheader`, every subsequent page will render that particular  cached header ie the one without the `extraheader` part.
To overcome this issue we need to make variants of the header (or any cached partial) depending on the `renderSearchIcon` Param (I renamed notRenderSearchIcon since it was not very reader friendly ).
Which lead me to the realisation that rather then manually adding header and footer in every new `section` (Contact, About,Newsletter etc) we should abstract the  include statements  in a block and reuse them rather then redefining them multiple times in other pages.
Your thoughts @gumaerc ?

File: base-theme/layouts/partials/header.html [hunk 1]
```diff
@@ -61,7 +63,9 @@
       </div>
     </div>
     <div class="right">
-      {{ block "extraheader" . }}{{ end }}
+      {{if ne (isset . "notRenderSearchIcon") true}}
```
Comment (gumaerc) [inline base-theme/layouts/partials/header.html:66]: This can be reverted too.

**Review by gumaerc — APPROVED** [review]
:+1: LGTM

**Pull request merged**